instrumentation/src/main/java/io/opentelemetry/android/internal/services/ServiceManager.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.services;
+
+import android.content.Context;
+import androidx.annotation.Nullable;
+import androidx.annotation.RestrictTo;
+import java.util.HashMap;
+
+public final class ServiceManager implements Lifecycle {
+
+    private final HashMap<Service.Type, Service> services = new HashMap<>();
```

Comment:
```suggestion
    private final Map<Service.Type, Service> services = new HashMap<>();
```

Is concurrency important to this class?

Reply:
> Is concurrency important to this class?

Not for now, ideally the writing to this map should happen once, during the rum initialization process.